FWCore/Framework/src/EventSetupRecordImpl.cc

```diff
@@ -96,8 +98,11 @@ EventSetupRecordImpl::add(const DataKey& iKey ,
       assert(proxy->providerDescription());
       assert(iProxy->providerDescription());
       if(iProxy->providerDescription()->isLooper_) {
-         (*proxies_.find(iKey)).second = iProxy ;
-	 return true;
+        proxies_[std::distance(keysForProxies_.begin(),
+                               std::lower_bound(keysForProxies_.begin(),
+                                                keysForProxies_.end(),
+                                                iKey))] = iProxy;
```

Comment:
I suppose here (and below where `std::lower_bound()` is used) `keysForProxies` is guaranteed to contain `iKey`?

Reply:
Yes, since the `find(iKey)` at line 87 did not return nullptr.